### ISSUE TITLE
[🐞] Dev mode failing since v0.20.0

### ISSUE THREAD

**Issue by EamonHeffernan** [BuilderIO/qwik#3321]
### Which component is affected?

Qwik Runtime

### Describe the bug

Since v0.20.0 dev mode fails with this error.

I am unable to reproduce the bug. however the message is

```
1:16:29 PM [vite] Internal server error: require() of ES Module C:\Users\Admin\Documents\Full-Stack\<project>\frontend\node_modules\@builder.io\qwik\build\index.mjs not supported.
Instead change the require of C:\Users\Admin\Documents\Full-Stack\<project>\frontend\node_modules\@builder.io\qwik\build\index.mjs to a dynamic import() which is available in all CommonJS modules.
      at Object.<anonymous> (C:\Users\Admin\Documents\Full-Stack\<project>\frontend\node_modules\@builder.io\qwik\core.cjs:17:13)
 ```

To get dev server to run (not without issues such as `Uncaught (in promise) SyntaxError: ambiguous indirect export: isServer`, modify `node_modules\@builder.io\qwik\build\package.json` and change `"main": "index.mjs",` to `"main": "index.cjs",`)

### Reproduction

.

### Steps to reproduce

pnpm dev

### System Info

```shell
System:
    OS: Windows 10 10.0.19045
    CPU: (16) x64 AMD Ryzen 7 3700X 8-Core Processor
    Memory: 3.86 GB / 15.93 GB
  Binaries:
    Node: 19.4.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.19 - C:\Program Files\nodejs\yarn.CMD
    npm: 9.2.0 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Spartan (44.19041.1266.0), Chromium (110.0.1587.63)
    Internet Explorer: 11.0.19041.1566
```

**Comment by doksara** [comment]
Any news on this?

**Comment by EamonHeffernan** [comment]
I poked around in debug mode and the issue seems to be caused by the module map for @qwik-serializer including an error. This was the `mod` object that vite generated that includes the error.
Sorry for the image format. I couldn't easily serialize it.

![image](https://user-images.githubusercontent.com/16983388/224605412-35cf662e-676e-45cb-8aa8-28f6df16b5be.png)

**Comment by EamonHeffernan** [comment]
Replacing the import for isServer with in platform.ts
`const isServer = /*#__PURE__*/ (() =>
  typeof process !== 'undefined' && !!process.versions && !!process.versions.node)();` makes the dev work with minimal errors. But I don't know where to go from here in terms of diagnosing the issue.

**Comment by manucorporat** [comment]
I would love to help debug this issue, but i dont know how to reproduce, a brand new created starter does not have this issue.

**Comment by NexonTreeHouse** [comment]
After updating to vite "4.2.0" (previously "3.2.4")  this error went away from me :)

**Comment by Alastair-smith2** [comment]
I found bumping vite also resolved the error above, I had been on `3.2.1` at the time of qwik `0.12.1`
